app/tasks/raster_source_assets.py

```diff
@@ -1,10 +1,70 @@
-from typing import Any, Dict
+import json
+from typing import Any, Dict, List
 from uuid import UUID
 
+from fastapi.encoders import jsonable_encoder
+
 from ..models.pydantic.change_log import ChangeLog
+from ..models.pydantic.creation_options import RasterTileSetSourceCreationOptions
+from ..models.pydantic.jobs import PixETLJob
+from ..settings.globals import ENV, S3_ENTRYPOINT_URL
+from . import Callback, callback_constructor, reader_secrets
+from .batch import execute
 
 
 async def raster_source_asset(
     dataset: str, version: str, asset_id: UUID, input_data: Dict[str, Any],
 ) -> ChangeLog:
-    pass
+
+    # pixETL does not currently support combining multiple inputs
+    source_uris: List[str] = input_data["creation_options"].get("source_uri", [])
+    if len(source_uris) > 1:
+        raise AssertionError("Raster sources only support one input file")
+    elif len(source_uris) == 0:
+        raise AssertionError("source_uri must contain a URI to an input file in S3")
+    source_uri = source_uris[0]
```

Comment:
I think something is still not quite right here.

1. If creating a raster asset from source, we expect creation_options.source_uri to have a value.
2. If creating from a default asset it will be None and we must infer the source (either point to database or to existing raster tile set in data lake).

The current test raises an error when source_uri is not set, which will always be the case under scenario 2

Reply:
I am also missing the part where we infer the source under scenario 2
- if default asset it geotable base table, set source_type to vector
- if default asset is raster tile set, set source_type to raster and point source_uri to tile.geojson in data lake